doc/guides/modules.rst

```diff
@@ -75,6 +75,65 @@ module:
      cmake: .
      kconfig: Kconfig
 
+
+Build system integration
+========================
+
+When a module has been described using the :file:`module.yml` file then it will
+automatically be included into the Zephyr Kconfig and CMake build system.
+
+To facilitate referencing between Zephyr and Zephyr modules all modules
+will be accessible through Kconfig and CMake variables.
+
+In Kconfig and CMake, the ``ZEPHYR_<module-name>_MODULE_DIR`` can be used to
+obtain the absolute path to the module directory.
+
+In CMake, the ``ZEPHYR_<module-name>_CMAKE_DIR`` can be used to obtain the
+absolute path to the :file:`CMakeLists.txt` that is included into CMake build
+system.
+
+To use the full path to Zephyr module ``FOO`` use:
+
+- Module directory in CMake: ``${ZEPHYR_FOO_MODULE_DIR}``
+- Module directory in Kconfig: ``$(ZEPHYR_FOO_MODULE_DIR)``
+- :file:`CMakeLists.txt` directory include in build system: ``${ZEPHYR_FOO_CMAKE_DIR}``
+
+Those variables may be used to obtain a full path to Zephyr module or to test
+whether a given module is present in the system.
+
+For example, to verify that ``FOO`` exists in the system and is a Zephyr module
+then one may do:
+
+.. code-block:: cmake
+
+  if(ZEPHYR_FOO_MODULE_DIR)
+    # Do something if FOO exists.
+  endif()
+
+In Kconfig, the variable may be used to specify a path to include.
+An example can be inclusion of additional Kconfig SoC files.
+
+In this example ``FOO`` specifies an additional ``SOC_ROOT`` and in
+:file:`<FOO>/soc/arm/foo_soc/Kconfig.soc` one may do:
+
+.. code-block:: kconfig
+
+  source "$(ZEPHYR_FOO_MODULE_DIR)/soc/arm/foo_soc/Kconfig.soc.common"
+
+During CMake processing of each Zephyr module, the following two variables are
+also available:
+
+- Module directory in CMake of current module being processed: ``${ZEPHYR_CURRENT_MODULE_DIR}``
+- :file:`CMakeLists.txt` directory of current module directory: ``${ZEPHYR_CURRENT_CMAKE_DIR}``
+
+This remove the need for a Zephyr module to know its own name during CMake
+processing, and the module can source additional CMake files using the current
+variable, for example:
```

Comment:
```suggestion
This removes the need for a Zephyr module to know its own name during CMake
processing. The module can source additional CMake files using these ``CURRENT``
variables. For example:
```

Reply:
@tejlmand it occurs to me that this means no module can be named "current".

Worth enforcing in the build system?

Reply:
Good catch. Let's hope someone doesn't create a current measurement sw tool, and call it `current`.

Reply:
added extra commit to check for `current`: https://github.com/zephyrproject-rtos/zephyr/pull/27740/commits/8511f9fd9a854c1963a415a6a5e93681eaf0d79b

Reply:
> added extra commit to check for `current`: [8511f9f](https://github.com/zephyrproject-rtos/zephyr/commit/8511f9fd9a854c1963a415a6a5e93681eaf0d79b)

nice :+1: